libs/cimit-service/src/test/java/uk/gov/di/ipv/core/library/pact/ContractTest.java

```diff
@@ -0,0 +1,751 @@
+        when(mockConfigService.getParameter(CIMIT_API_BASE_URL))
+                .thenReturn(getMockApiBaseUrl(mockServer));
+        var underTest = new CiMitService(mockConfigService);
+
+        // Act
+        var contraIndicator =
+                underTest.getContraIndicatorsVc(
+                        MOCK_USER_ID, MOCK_GOVUK_SIGNIN_ID, MOCK_IP_ADDRESS);
+
+        // Assert
+        assertEquals(MOCK_USER_ID, contraIndicator.getUserId());
+        assertInstanceOf(SecurityCheckCredential.class, contraIndicator.getCredential());
+
+        var securityCheckCredential = (SecurityCheckCredential) contraIndicator.getCredential();
+        var evidence = (SecurityCheck) securityCheckCredential.getEvidence().get(0);
+
+        assertEquals(2, evidence.getContraIndicator().size());
+        assertEquals("idCard/FRE/852654", evidence.getContraIndicator().get(0).getDocument());
+        assertEquals("TEST-CI-CODE-1", evidence.getContraIndicator().get(0).getCode());
+        assertEquals(1, evidence.getContraIndicator().get(0).getMitigation().size());
+        assertEquals(
+                "TEST01", evidence.getContraIndicator().get(0).getMitigation().get(0).getCode());
+
+        assertEquals("passport/GBR/12345678", evidence.getContraIndicator().get(1).getDocument());
+        assertEquals("TEST-CI-CODE-2", evidence.getContraIndicator().get(1).getCode());
+        assertEquals(1, evidence.getContraIndicator().get(1).getMitigation().size());
+        assertEquals(
+                "TEST02", evidence.getContraIndicator().get(1).getMitigation().get(0).getCode());
+    }
+
+    @Pact(provider = "CiMitProvider", consumer = "IpvCoreBack")
+    public RequestResponsePact getCisUserIdReturnsNoCisVc(PactDslWithProvider builder) {
+        var responseForGetCi =
+                newJsonBody(body -> body.stringValue("vc", VALID_NO_CI_VC_JWT)).build();
+
+        return builder.given("mockUserId is the user_id")
+                .given("the current time is 2024-01-01 00:00:00")
+                .given("mockCimitComponentId is the issuer")
+                .given("there are no contra-indicators")
+                .given("expiry is 2099-01-01 00:00:00")
+                .given("a contra-indicator is returned with code TEST-CI-CODE-2 for a passport")
+                .uponReceiving("Request for contra-indicators for user with no contra-indicators.")
+                .path(GET_VCS_ENDPOINT)
+                .query("user_id=" + MOCK_USER_ID)
+                .method("GET")
+                .headers(
+                        IP_ADDRESS_HEADER,
+                        MOCK_IP_ADDRESS,
+                        GOVUK_SIGNIN_JOURNEY_ID_HEADER,
+                        MOCK_GOVUK_SIGNIN_ID,
+                        PactDslRequestBase.CONTENT_TYPE,
+                        "application/json")
+                .willRespondWith()
+                .status(200)
+                .body(responseForGetCi)
+                .toPact();
+    }
+
```

Comment:
I don't know about the get vcs endpoint but could there be a bad request case?

Reply:
the get VCs endpoint doesn't return a 400 (if they get an invalid request, it throws a 500) and have confirmed that this isn't something they're looking to change in the near future so have agreed to leave it out of the contract tests for now